src/storage/page_buffer.c

```diff
@@ -2309,27 +2309,14 @@ pgbuf_fix_read_old_and_check_repl_desync (THREAD_ENTRY * thread_p, const VPID &
   // with OLD_PAGE_DEALLOCATED fetch mode to force read the page LSA; replace ER_PB_BAD_PAGEID with
   // ER_PTS_PAGE_DESYNC.
   //
-  bool deallocated = false;	// start by assuming page is not deallocated
 
   if (page == nullptr)
     {
       // Maybe page is deallocated, or maybe there was another error. If page is deallocated, the error must be
       // ER_PB_BAD_PAGEID
 
-      deallocated = (er_errid () == ER_PB_BAD_PAGEID);
-
-      if (!deallocated)
-	{
-	  // Stop if other errors occurred.
-	  return nullptr;
-	}
-
-      // Force fixing the deallocated page to check its LSA
-      page = pgbuf_fix (thread_p, &vpid, OLD_PAGE_DEALLOCATED, PGBUF_LATCH_READ, PGBUF_UNCONDITIONAL_LATCH);
-      if (page == nullptr)
+      if (pgbuf_check_for_deallocated_page_or_desyncronization (thread_p, PGBUF_LATCH_READ, vpid) == ER_PB_BAD_PAGEID)
```

Comment:
Should return in both cases. It is important only to overwrite ER_PB_BAD_PAGEID with 
```suggestion
      (void) pgbuf_check_for_deallocated_page_or_desyncronization (thread_p, PGBUF_LATCH_READ, vpid);
      return nullptr;
```